app/views/tasks/index.html.erb

```diff
@@ -33,33 +33,37 @@
     </div>
 
     <!-- Select category and add task-->
-    <div class="d-flex align-items-center justify-content-between px-3 py-1">
-      <div class="d-flex align-items-center">
-        <small class="me-1 my-2">Filter tasks by</small>
-        <select class="border-0 bg-light rounded text-secondary">
-          <option selected>None</option>
-          <option>Inventory</option>
-        </select>
+    <div class = "d-flex align-items-center justify-content-between px-3 py-1">
+      <div id = "filter-category-dropdown"class = "d-flex align-items-center">
+        <small class = "me-1 my-2">Filter tasks by</small>
+        <label>
+          <select onchange = "location = this.value;" id="category-filter" class = "border-0 bg-light rounded text-secondary">
+            <option value = "<%= tasks_path %>" <%= params[:category].blank? ? 'selected' : ''%>>All Categories</option>
+            <% @curr_categories.each do |category| %>
+              <option value = "<%= tasks_path(:category => category) %>" <%= params[:category] == category ? 'selected' : ''%>><%= category %></option>
+            <% end %>
+          </select>
+        </label>
       </div>
       <button type="button" class="btn btn-sm btn-danger" data-bs-toggle="modal" data-bs-target="#add-task-modal">Add Task +</button>
     </div>
   </div>
 </div>
 
-<!-- Modal -->
-<div class="modal fade" id="add-task-modal" tabindex="-1" aria-labelledby="add-task-modal" aria-hidden="true">
-  <div class="modal-dialog">
-    <div class="modal-content">
-      <div class="modal-header">
-        <h5 class="modal-title">Add New Task</h5>
-        <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+<!-- Add Task Modal -->
+<div class = "modal fade" id = "add-task-modal" tabindex = "-1" aria-labelledby = "add-task-modal" aria-hidden = "true">
+  <div class = "modal-dialog">
+    <div class = "modal-content">
+      <div class = "modal-header">
+        <h5 class = "modal-title">Add New Task</h5>
+        <button type = "button" class = "btn-close" data-bs-dismiss = "modal" aria-label = "Close"></button>
       </div>
-      <div class="modal-body">
-        <div class="d-flex justify-content-end">
-            <%= image_tag "bsfc_logo.png", {:alt => "Berkeley Student Food Collective Logo", :width => 100, :height => 100} %>
+      <div class = "modal-body">
+        <div class = "d-flex justify-content-center">
+          <%= image_tag "bsfc_logo.png", {:alt => "Berkeley Student Food Collective Logo", :width => 100, :height => 100} %>
         </div>
-        <%= form_tag tasks_path, :method => "post", :class => "" do %>
-          <div class="mb-2">
+        <%= form_tag tasks_path(:category => params[:category]), :method => "post", :class => "" do %>
+          <div class = "mb-2">
```

Comment:
please clean this up - spaces around = for attributes are highly deprecated. it was correct before